src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -162,6 +169,56 @@ static ImmutableArray<IMethodSymbol> GetOperatingSystemGuardMethods(IMethodSymbo
                     (method.Parameters.Length == 0 || method.Name.EndsWith(OptionalSuffix, StringComparison.Ordinal));
         }
 
+        private static bool PlatformAnalysisAllowed(AnalyzerOptions options, Compilation compilation, CancellationToken token)
+        {
+            var tfmString = options.GetMSBuildPropertyValue(MSBuildPropertyOptionNames.TargetFramework, compilation, token);
+
+            if (tfmString != null &&
+               TryParseTfm(tfmString, out var platform, out var version))
+            {
+                if (platform.Equals(Net, StringComparison.OrdinalIgnoreCase) &&
+                    version != null &&
+                    int.TryParse(version[0].ToString(), out var value) &&
+                    value >= 5)
+                {
+                    return true;
+                }
+                else
+                {
+                    return LowerTargetsEnabled(options, compilation, token);
+                }
+            }
+
+            return false;
+        }
+
+        private static bool LowerTargetsEnabled(AnalyzerOptions options, Compilation compilation, CancellationToken cancellationToken) =>
+            compilation.SyntaxTrees.FirstOrDefault() is { } tree &&
+            options.GetBoolOptionValue(EditorConfigOptionNames.EnablePlatformAnalyzer, SupportedOsRule, tree, compilation, false, cancellationToken);
+
+        private static bool TryParseTfm(string tfm, [NotNullWhen(true)] out string? platform, out string? version)
+        {
+            version = null;
+            for (int i = 0; i < tfm.Length; i++)
+            {
+                if (char.IsDigit(tfm[i]))
+                {
+                    if (i > 0)
+                    {
+                        platform = tfm.Substring(0, i);
+                        version = tfm[i..];
```

Comment:
Will this handle all flavored platforms mentioned here: https://github.com/dotnet/designs/blob/master/accepted/2020/net5/net5.md#design?

Reply:
Yes, except `Tizen, Unity` i guess they are not expected to be supported though. Luckily we don't need to care about the different platforms, only `net5` section is matter from `net5.0-XYZ` so we are only checking if everything `before the first number` is equal to `net` and the `first number` is >= 5

Reply:
I thought the range expression will give entire substring starting from index i till the end.

Reply:
Can we simplify the check to:
1. .Length >= 4 AND
2. .StartsWith(“net”, OrdinalIgnoreCase) AND
3 int.TryParse(tfm[3], out int major) AND
4. major >= 5

Reply:
Yes, we don't really need to parse entire TFM